### PR TITLE
Zero add unsafe mpoly

### LINKED ISSUE #732: No method matching zero!(::fmpz_mpoly)
The last line of the following causes an error, I think this must be a bug::

```julia
using Nemo
R, (a, b) = PolynomialRing(ZZ, ["a", "b"])
T, x = PolynomialRing(R, "x")
divexact(x, one(T))
U, (y,) = PolynomialRing(R, ["y"])
divexact(y, one(U))
```

I believe I used the latest version of Nemo (installed with ]dev Nemo).

Anders

### PR DESCRIPTION
**Pull request by wbhart**
Closes #732

### CODE REVIEW

File: src/flint/fmpq_mpoly.jl [hunk 1]
```diff
@@ -725,6 +725,19 @@ end
...
 ###############################################################################
 
+function zero!(a::fmpq_mpoly)
+    ccall((:fmpq_mpoly_zero, :libflint), Nothing,
+         (Ref{fmpq_mpoly}, Ref{FmpqMPolyRing}), a, a.parent)
+    return a
+end
+
+function add!(a::fmpq_mpoly, b::fmpq_mpoly, c::fmpq_mpoly)
+   ccall((:fmpz_mpoly_add, :libflint), Nothing,
```
Comment (tthsqe12) [inline src/flint/fmpq_mpoly.jl]: fmpq_mpoly_add!

File: src/flint/nmod_mpoly.jl [hunk 2]
```diff
@@ -696,6 +696,19 @@ end
...
 ###############################################################################
 
+function zero!(a::nmod_mpoly)
+    ccall((:nmod_mpoly_zero, :libflint), Nothing,
+         (Ref{nmod_mpoly}, Ref{NmodMPolyRing}), a, a.parent)
+    return a
+end
+
+function add!(a::nmod_mpoly, b::nmod_mpoly, c::nmod_mpoly)
+   ccall((:fmpz_mpoly_add, :libflint), Nothing,
```
Comment (tthsqe12) [inline src/flint/nmod_mpoly.jl]: nmod_mpoly_add!

**Comment by tthsqe12** [comment]
Test code insufficient.

**Comment by wbhart** [comment]
Oh boy. HOW did I miss that!? I went over the code a dozen times, corrected it about 5 times, looking for precisely things like this, because.... we don't directly test the unsafe operators at the moment.

**Comment by thofma** [comment]
Hmm, were the tests picking this up? Did not see an error after the first two commits.

**Comment by wbhart** [comment]
@thofma We don't test the unsafe operators explicitly anywhere, I think.

**Comment by thofma** [comment]
Yes, right. I would have guessed that they are called implicitely somewhere in the tests, but now I realize that this would only happen when doing for example polynomial arithmetic or matrix arithmetic over the multivariate polynomial ring.

**Pull request merged**